### PR TITLE
feat: manage groups from deployment settings for single-org deployments

### LINKED ISSUE #13957: Integrate all users view into the multi-org capable `ManagementSettingsPage`
### Notes

- The "Users" sidebar item currently just directs you to the existing /users route.
- Ensure that groups can still be modified for single-org deployments. (We've been exposing groups as a top-level resource, but they're actually org scoped, so we need to make sure these settings stay accessible)
- Don't show organization settings for single-org deployments. All users should be added as members of the single-org transparently, and the org-level auditing page would just show all the same things as the global one.
- Groups should be shown as a "Deployment" setting rather than under an org, until there is more than one org.

### Goals

- [x] Change the /users route to /deployment/users, and add a `<Redirect>` to avoid breaking links.
- [x] Make sure that the page integrates with/looks good in the new context (eg. the correct sidebar item is highlighted).
- [x] Remove the Users/Groups tab bar from the top of the Users page
- [x] Add a "Groups" sidebar item underneath the Deployment settings for single-org deployments.
- [ ] Add a /deployment/groups route that behaves the same as /organizations/default/groups, and redirects if there is more than one org.

### PR DESCRIPTION
**Pull request by aslilac**
Closes #13957

Screenshot 2024-07-25 at 12 30 46 PM

### CODE REVIEW

**Review by code-asher — COMMENTED** [review]
Just to make sure I understand "single-org" in this context, that is with the multi-org experiment enabled but not licensed, right?

Should we redirect `/groups` to `/deployment/groups` (or `/organizations/default/groups/` for multi-org?) similar to the users page?  Probably not very important, but if someone has a bookmark or tries typing it manually maybe.

I think the create group page needs to take me to `/deployment/groups/groupname` rather than `/organizations/default/groups/groupname` with single-org since the latter gives me a blank sidebar (maybe a good place for another relative redirect).

Looks good so far!

File: site/src/api/queries/groups.ts [hunk 1]
```diff
@@ -6,18 +6,18 @@ import type {
...
+  "organization",
   organizationId,
   "group",
   groupName,
 ];
 
 export const groups = (organizationId: string) => {
   return {
-    queryKey: GROUPS_QUERY_KEY,
+    queryKey: ["organization", organizationId, "groups"],
```
Comment (BrunoQuaresma) [inline site/src/api/queries/groups.ts:20]: I see that this query key is being used in multiple places. Should we centralize it to ensure consistency?

File: site/src/pages/ManagementSettingsPage/GroupsPage/GroupPage.tsx [hunk 2]
```diff
@@ -50,7 +50,7 @@ import { isEveryoneGroup } from "utils/groups";
 import { pageTitle } from "utils/page";
 
 export const GroupPage: FC = () => {
-  const { organization, groupName } = useParams() as {
+  const { organization = "default", groupName } = useParams() as {
```
Comment (BrunoQuaresma) [inline site/src/pages/ManagementSettingsPage/GroupsPage/GroupPage.tsx:53]: I see the constant "default" used in multiple places. We should define it in a `const.ts` file for consistency. Makes sense?

**Comment by BrunoQuaresma** [comment]
1. When I go to Deployment > Organizations, the sidebar is hidden. It looks strange to me; I would prefer to see it open.

Screenshot 2024-07-26 at 10 09 55

2. It is not related to this particular PR, but I've been finding it hard to determine the currently active menu in the sidebar due to the numerous options available. Maybe we could style the active option differently?

Screenshot 2024-07-26 at 10 11 26

3. I have a license and multi-org enabled, but I can't create new orgs. Perhaps it's related to 1?
Screenshot 2024-07-26 at 10 13 40
Screenshot 2024-07-26 at 10 14 40
Screenshot 2024-07-26 at 10 13 49

**Review by BrunoQuaresma — COMMENTED** [review]
I'm missing tests for the changes so we can be sure we are not introducing bugs while moving forward.

File: site/src/pages/ManagementSettingsPage/GroupsPage/GroupPage.tsx (hunk 2 above)
Comment (aslilac) [reply site/src/pages/ManagementSettingsPage/GroupsPage/GroupPage.tsx:53]: personally, I really dread adding imports like that which end up being more typing that the thing you could just say directly

**Comment by aslilac** [comment]
> I think it is also true when there is only one org created.

No, a deployment which only has one org created but has a multi-org license is still a multi-org deployment. We need to show the "New organization" button, for example which is only present in the multi-org version of the UI. (and maintaining a 3rd version of the UI that's half single-org and half multi-org would be a gigantic pain)

**Comment by aslilac** [comment]
> I have a license and multi-org enabled, but I can't create new orgs.

there are two license types now, and old/enterprise licenses no longer have access to org features. you can get a "premium" one from notion to test with. :)

**Review by BrunoQuaresma — APPROVED** [review]

**Pull request merged**